File: onDeck/Managers/ScheduleManager.swift
import Foundation

@Observable
@MainActor
final class ScheduleManager {
    var todaysGames: [Game] = []
    var error: String?

    private let mlbAPI = MLBStatsAPI()

    /// Fetches today's schedule and filters to games involving the given team names.
    func fetchSchedule(for teamNames: Set<String>) async {
        error = nil
        do {
            let allGames = try await mlbAPI.fetchSchedule(date: .now)
            todaysGames = allGames.filter { game in
                teamNames.contains(game.homeTeam) || teamNames.contains(game.awayTeam)
            }
        } catch {
            self.error = "Schedule fetch failed: \(error.localizedDescription)"
        }
    }

    /// Re-fetches on day rollover. Call this at midnight or when the app becomes active.
    func refreshIfNeeded(teamNames: Set<String>) async {
        await fetchSchedule(for: teamNames)
    }
}

File: windows/src/OnDeck.Core/Managers/ScheduleManager.cs
using OnDeck.Core.Models;
using OnDeck.Core.Networking;
using OnDeck.Core.Utilities;

namespace OnDeck.Core.Managers;

/// <summary>Port of <c>Managers/ScheduleManager.swift</c>.</summary>
public sealed class ScheduleManager(MlbStatsApi mlb, TimeProvider? timeProvider = null)
{
    private readonly TimeProvider _time = timeProvider ?? TimeProvider.System;

    public IReadOnlyList<Game> TodaysGames { get; private set; } = [];

    public string? Error { get; private set; }

    /// <summary>
    /// Fetches today's schedule and filters to games involving the given team names.
    /// Uses the "baseball day" — before 8 AM counts as the previous day.
    /// </summary>
    public async Task FetchScheduleAsync(IReadOnlySet<string> teamNames, CancellationToken ct = default)
    {
        Error = null;

        try
        {
            var allGames = await mlb.FetchScheduleAsync(BaseballCalendar.Today(_time), ct);
            TodaysGames =
            [
                .. allGames.Where(game =>
                    teamNames.Contains(game.HomeTeam) || teamNames.Contains(game.AwayTeam))
            ];
        }
        catch (Exception ex)
        {
            Error = $"Schedule fetch failed: {ex.Message}";
        }
    }
}
